docs/sql-ref-syntax-qry-select-tvf.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: global
+title: Table-valued Functions (TVF)
+displayTitle: Table-valued Functions (TVF)
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+A table-valued function (TVF) is a function that returns a relation or a set of rows.
+
+### Syntax
+
+{% highlight sql %}
+function_name ( expression [ , ... ] ) [ [ AS ] table_alias ]
+{% endhighlight %}
+
+### Parameters
+
+<dl>
+  <dt><code><em>expression</em></code></dt>
+  <dd>
+    Specifies a combination of one or more values, operators and SQL functions that results in a value.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>table_alias</em></code></dt>
+  <dd>
+    Specifies a temporary name given to a table.
+  </dd>
+</dl>
+
+### Examples
+
+{% highlight sql %}
+-- range call with end
+SELECT * FROM range(6 + cos(3));
++---+
+| id|
++---+
+|  0|
+|  1|
+|  2|
+|  3|
+|  4|
++---+
+
+-- range call with start and end
+SELECT * FROM range(5, 10);
++---+
+| id|
++---+
+|  5|
+|  6|
+|  7|
+|  8|
+|  9|
++---+
+
+-- range call with numPartitions
+SELECT * FROM range(0, 10, 1, 200);
```

Comment:
I think we need to explain `range` parameters, (end, start, step, numPartitions).

Reply:
How about adding a table explaining implemented TVFs just like https://www.postgresql.org/docs/current/functions-srf.html ?

Reply:
Sounds good. Besides range, what other TVFs Spark supports?

Reply:
Yea, yes, See: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala#L83